test/bcreader_tests.jl

```diff
@@ -0,0 +1,308 @@
+        interpolate_daily = true
+        dummy_dates = Vector(range(DateTime(1999, 1, 1); step = Day(1), length = 100))
+        segment_idx0 = [Int(1)]
+
+        # these values give an `interp_fraction` of 0.5 in `interpol` for ease of testing
+        date0 = dummy_dates[Int(segment_idx0[1] + 1)]
+        segment_length = [Int(2) * ((date0 - dummy_dates[Int(segment_idx0[1])]).value)]
+
+        radius = FT(6731e3)
+        Nq = 4
+        domain = Domains.SphereDomain(radius)
+        mesh = Meshes.EquiangularCubedSphere(domain, 4)
+        topology = Topologies.Topology2D(mesh)
+        quad = Spaces.Quadratures.GLL{Nq}()
+        boundary_space_t = Spaces.SpectralElementSpace2D(topology, quad)
+        monthly_fields = (zeros(boundary_space_t), ones(boundary_space_t))
+
+        bcf_info_interp = BCReader.BCFileInfo(
+            FT,                                 # FT
+            "",                                 # bcfile_dir
+            ClimaComms.SingletonCommsContext(), # comms_ctx
+            "",                                 # hd_outfile_root
+            "",                                 # varname
+            dummy_dates,                        # all_dates
+            deepcopy(segment_idx0),             # segment_idx
+            segment_idx0,                       # segment_idx0
+            monthly_fields,                     # monthly_fields
+            segment_length,                     # segment_length
+            nothing,                            # scaling_function
+            interpolate_daily,                  # interpolate_daily
+            nothing,                            # land_mask
+        )
+        @test BCReader.interpolate_midmonth_to_daily(FT, date0, bcf_info_interp) == ones(boundary_space_t) .* FT(0.5)
+
+        # test interpolate_midmonth_to_daily without interpolation
+        interpolate_daily = false
+
+        bcf_info_no_interp = BCReader.BCFileInfo(
+            FT,                                 # FT
+            "",                                 # bcfile_dir
+            ClimaComms.SingletonCommsContext(), # comms_ctx
+            "",                                 # hd_outfile_root
+            "",                                 # varname
+            dummy_dates,                        # all_dates
+            deepcopy(segment_idx0),             # segment_idx
+            segment_idx0,                       # segment_idx0
+            monthly_fields,                     # monthly_fields
+            segment_length,                     # segment_length
+            nothing,                            # scaling_function
+            interpolate_daily,                  # interpolate_daily
+            nothing,                            # land_mask
+        )
+        @test BCReader.interpolate_midmonth_to_daily(FT, date0, bcf_info_no_interp) == monthly_fields[1]
+    end
+
+    # Add tests which use TempestRemap here -
```

Comment:
Don't the tests below use TR?

Reply:
The two tests below this both call `bcfile_info_init`, which uses TR so it can't be run on windows